js/app/historyStore.js

```diff
@@ -91,9 +92,11 @@ const HistoryStore = new Lang.Class({
         Dispatcher.get_default().register((payload) => {
             switch(payload.action_type) {
                 case Actions.HISTORY_BACK_CLICKED:
+                    this.get_previous_item().entry_point = EntryPoints.NAV_BUTTON_CLICKED;
```

Comment:
Right, you don't have to make duplicate calls, but the alternative is kind of messy. I pushed a commit to show what I mean, but I couldn't test it because of some gjs issues.